src/views/Graph.vue

```diff
@@ -508,12 +837,81 @@ export default {
      * @returns {{ [dateTime: string]: Object[] }=} mapping of cycle points to nodes
      */
     getCycles (nodes) {
-      if (!this.groupCycle) return
       return nodes.reduce((x, y) => {
         (x[y.tokens.cycle] ||= []).push(y)
         return x
       }, {})
     },
+    addSubgraph (dotcode, pointer, graphSections) {
+      pointer.children.forEach((key, i) => {
+        const value = key
+        const children = this.allChildrenLookUp[value.id]
+        if (!children) { return }
+        const removedNodes = []
+        children.forEach((a) => {
+          if (this.collapseFamily.includes(a.name)) {
+            a.children.forEach((child) => {
+              removedNodes.push(child.name)
+            })
+          }
+        })
+        // filter parent
+        if (
+          children.length &&
+          this.groupFamily.includes(key.node.name) &&
+          !this.collapseFamily.includes(key.node.name) &&
+          !this.collapseFamily.includes(key.node.firstParent.name)) {
+          // filter child
+          const nodeFormattedArray = children.filter((a) => {
+            // if its not in the list of families (unless its been collapsed)
+            const isFamily = !this.familyArrayStore.includes(a.name) || this.collapseFamily.includes(a.name)
+            // its the node has been removed/collapsed
+            const isRemoved = !removedNodes.includes(a.name)
+            // is not numeric
+            const isNumeric = !parseFloat(a.name)
+            return isFamily && isRemoved && isNumeric
+          }).map(a => `"${a.id}"`)
+          if (nodeFormattedArray.length) {
+            dotcode.push(`
+            subgraph cluster_margin_family_${key.name}${key.tokens.cycle}
+              {
+              margin=100.0
+              label="margin"
+              subgraph cluster_${key.name}${key.tokens.cycle}
+                {${nodeFormattedArray}${nodeFormattedArray.length ? ';' : ''}
+                  label = "${key.name}"
+                  fontsize = "70px"
+                  style=dashed
+                  margin=60.0
+            `)
+          }
+        }
+
+        if (value) {
+          this.addSubgraph(dotcode, value, graphSections)
+        }
+        if (Object.keys(graphSections).includes(key)) {
+          dotcode.push(graphSections[key.id])
+        }
+
+        if (
+          children.length &&
+          this.groupFamily.includes(key.node.name) &&
+          !this.collapseFamily.includes(key.node.name) &&
+          !this.collapseFamily.includes(key.node.firstParent.name)) {
+          const nodeFormattedArray = children.filter((a) => {
+            // if its not in the list of families (unless its been collapsed)
+            const isFamily = !this.familyArrayStore.includes(a.name) || this.collapseFamily.includes(a.name)
+            // its the node has been removed/collapsed
+            const isRemoved = !removedNodes.includes(a.name)
+            return isFamily && isRemoved
```

Comment:
This block looks identical to the one above (only without the isNumeric check).

So I think we can remove it completely like this:

```diff
diff --git a/src/views/Graph.vue b/src/views/Graph.vue
index 9973a43a..fa070fb2 100644
--- a/src/views/Graph.vue
+++ b/src/views/Graph.vue
@@ -855,7 +855,9 @@ export default {
             })
           }
         })
+
         // filter parent
+        let openedBrackets = false
         if (
           children.length &&
           this.groupFamily.includes(key.node.name) &&
@@ -872,6 +874,7 @@ export default {
             return isFamily && isRemoved && isNumeric
           }).map(a => `"${a.id}"`)
           if (nodeFormattedArray.length) {
+            openedBrackets = true
             dotcode.push(`
             subgraph cluster_margin_family_${key.name}${key.tokens.cycle}
               {
@@ -894,21 +897,8 @@ export default {
           dotcode.push(graphSections[key.id])
         }
 
-        if (
-          children.length &&
-          this.groupFamily.includes(key.node.name) &&
-          !this.collapseFamily.includes(key.node.name) &&
-          !this.collapseFamily.includes(key.node.firstParent.name)) {
-          const nodeFormattedArray = children.filter((a) => {
-            // if its not in the list of families (unless its been collapsed)
-            const isFamily = !this.familyArrayStore.includes(a.name) || this.collapseFamily.includes(a.name)
-            // its the node has been removed/collapsed
-            const isRemoved = !removedNodes.includes(a.name)
-            return isFamily && isRemoved
-          }).map(a => `"${a.id}"`)
-          if (nodeFormattedArray.length) {
-            dotcode.push('}}')
-          }
+        if (openedBrackets) {
+          dotcode.push('}}')
         }
       })
     },
```